pages/download.vue

```diff
@@ -88,32 +109,103 @@
 
         mounted() {
 
-            // Set the current page
+            // 1. Set the current page
             this.$store.dispatch("setCurrentPage", "download");
+
+            // 2. Create a reverse of the column to category map in the store
+            this.refreshCategoryToColumnMap();
         },
 
         methods: {
 
-            saveFile() {
-
-                // 1. Create a blob out of the annotated table data
-                const data = JSON.stringify(this.dataTable.annotated);
-                const blob = new Blob([data], {type: "text/plain"});
-
-                // 2. Create an anchor tag in memory linked to the blob
-                const pseudoAnchor = document.createElement("a");
-                pseudoAnchor.download = "annotated_data.json";
-                pseudoAnchor.href = window.URL.createObjectURL(blob);
-                pseudoAnchor.dataset.downloadurl = [
-                    "text/json",
-                    pseudoAnchor.download,
-                    pseudoAnchor.href
-                ].join(':');
-
-                // 3. Dispatch a mouse click event on the in-memory anchor tag
-                const pseudoClickEvent = document.createEvent("MouseEvents");
-                pseudoClickEvent.initEvent("click", true, false, window, 0, 0, 0, 0, 0, false, false, false, false, 0, null);
-                pseudoAnchor.dispatchEvent(pseudoClickEvent);
+            downloadAnnotatedData() {
+
+                // 1. Format the annotated table into propietary JSON format
+                const jsonData = this.transformAnnotatedTableToJSON();
+
+                // 2. Open file dialog to prompt the user to name it and
+                // download it to their location of choice
+                this.fileSaverSaveAs(jsonData);
+            },
+
+            fileSaverSaveAs(p_jsonData) {
+
+                var blob = new Blob([JSON.stringify(p_jsonData, null, this.jsonSpacing)], {type: "text/plain;charset=utf-8"});
+                saveAs(blob, this.defaultOutputFilename);
+            },
+
+            refreshCategoryToColumnMap() {
```

Comment:
I'm pretty sure this is a duplicate from elsewhere in the app, no?
Let's at least add a `// TODO` flag or something like it so we don't forget to refactor this!

Reply:
No, this is in fact a method to create/refresh a reverse of the column to category map in the store. `categoryToColumnMap` is a local object only used by the download page. Made sense to update it in some way in case a user altered the `columnToCategoryMap` in the store.